ecosystem/indexer-grpc/indexer-grpc-table-info/src/table_info_parser.rs

```diff
@@ -93,7 +92,27 @@ impl TableInfoParser {
             tasks.push(task);
         }
         match futures::future::try_join_all(tasks).await {
-            Ok(res) => res,
+            Ok(res) => {
+                if !db.reader.is_indexer_async_v2_pending_on_empty().unwrap() {
+                    info!("Missing data in table info parsing at start version {:?}, retrying now", self.current_version);
+
+                    let batch_to_reprocess = TransactionBatchInfo {
+                        start_version: self.current_version,
+                        num_transactions_to_fetch: self.parser_batch_size *self.parser_task_count,
+                    };
+                    let context = self.context.clone();
+                    let ledger_version = self.highest_known_version;
+                    let raw_txns =
+                    Self::fetch_raw_txns_with_retries(context.clone(), ledger_version, batch_to_reprocess).await;
+                    Self::parse_table_info(context.clone(), raw_txns.clone(), db_writer.clone())
+                        .expect("Failed to parse table info in retry");
```

Comment:
Can we just rerun this? Why do we need to fetch again?

Reply:
Also I guess this wouldn't be efficient b/c you have to run through every transaction again. Can we run it just on the specific handles? E.g. https://github.com/aptos-labs/aptos-core/blob/b00c6868eb51998e3b53d25cd4f3df35e2c5509b/storage/indexer/src/lib.rs#L268-L272